### DISCUSSION TITLE
[Enhancement] [button] [Component][button][link] Make clear the intention of the library when it comes to links that look like a button and have an el-link button='true' in the same way we have el-button link='true'

### DISCUSSION THREAD

**Discussion by thebravoman** [element-plus/element-plus#11359] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `button`

## Description

It is not clear in Element Plus documentation how links that look like buttons could be implemented. This could be improved in the documentation as it is kind of ambiguous, not clear, and might lead to the library being used in cases that generate invalid HTML. 

In a normal well designed web page a button will do an action and a link will take us to a resource. Element Plus web site follow this convention and links take us to new pages and buttons send some information to the server. 

But when moving to element plus we often do it from other libraries that had different conventions - like displaying a link that looks like a button but behaves like a link

Generally this is achieve (in bootstrap) for example with 

```html
<a class="btn btn-primay">
```

This will produce a link in bootstrap that looks like button which, I will admit, some designers and users love a lot. Whether this should be like this or not is a choice of the designer and their vision for the UI. 

A good example is a "New Article" button on a page. When creating a new article we would like to take the user to a new page (not open a modal) and gather the information that we need for the New Article. 

In this case New Article takes us from "/articles" to "/articles/new". So New Article is a link in that sense that it takes us to a new page. But in the sense of the UI the user is taking an action asking the server to create a new article and it makes sense for it to be a button. The server can not create the new article, but it could open a new article page. 

Element Plus documentation is not clear about how could this be achieved - to have a link that looks like a button.

I believe an library like Element Plus should provide this "tool" to developer and designer. In 2.2.1 there is way to make a button look like a link by using the link='true' API.

The code

```html
<el-button type="primary" link="true">
```
generates an HTML button element that looks like a link.

But there is no way to generate a link that looks like a button like

```html
<el-link button="true">
```

In the same time:

```html
<el-button type="primary"> 
```
generates an HTML button element that looks like a button but does not have a link.

Naturally there are four ways in the current state of Element Plus where we can achieve "a link that looks like a button"

1. Create an el-button inside of an 'a' element like this

```html
<a href="...">
  <el-button type="primary"...>New Campaign</el-button>
</a>
```

This disadvantage of this approach is that it generates an invalid HTML. Buttons can not be inside in 'a' tags. The W3C validator helpfully tells us that.

```html
Error: The element button must not appear as a descendant of the a element.

From line 2, column 3; to line 2, column 10

://dd">↩  <button>Text</
```
(result by https://validator.w3.org)

This have implications. There are tools - in our case a tool for automated testing that helpfully inform us that there are two elements the use can click on. The two elements are the A and the Button and both are with the same text. This is not what we want from the user interface. The challenges with generating invalid HTML are also around supporting it. We never know what and how will break down the road. 

2. Create an el-link with el-button 

Instead of 

```html
<a><el-button></el-button></a> 
```
we can use the Element Plus el-link 

```html
<el-link><el-button></el-button></el-link>
```
This follows the same disadvantages as the one above. It is a button inside of a link

3. Create an 'el-link' with a class='el-button'

```html
<el-link class='el-button'>
```

Now in this way we have only a link in the HTML. Which is great. And it looks like a button. Which is also great because this is how the designers and the users want it. (Note, even If I personally like to use buttons to take action with a server and links to go to new pages, this is not what many users and designers expect. People kind of got used to having things that look like buttons but behave like links)

4. Directly use "a class='el-button'"

In this way we use the HTML anchor tag directly. And it looks like a button.

There is also a deep issue with inconsistency in the library. We can do 

```html
<el-button link="true">
```

but there is no way to do 

```html
<el-link button="true">
```

In summary:

My suggestion with this feature requests is for guidance in Element Plus documentation on how should this be achieved using this marvelous library and probably the adding a property button="true" to 'el-link.

Let me know your thoughts.

### Additional comments

_(empty)_

**Comment by tolking** [comment]
same with #11324

Maybe we can add an attribute to define what tag the ElButton component renders.

**Reply by thebravoman** [reply]
It is the same.

Adding an attribute to define the tag might work. 

But then it will strange for the newly introduced link=true in el-button

Imagine

```
<el-button link=true tag=button>
```

Because we are telling it to use el-button, to generate a link with link=true but to use a 'button' tag because of tag=button.

A much more consistent approach with 

```
<el-button link=true> # This one we have
<el-link button=true> # This one could be introduced and consistent with the above one
```

The second one is currently not available.

**Reply by tolking** [reply]
The `<el-link button=true>` is not very compliant with the design specifications.

**Reply by thebravoman** [reply]
Help me understand, what are these design specifications?

And how is 
```
<el-button link=true>
``` 
compliant with them but 
```
<el-link button=true>
```

 is not?

**Reply by tolking** [reply]
I mean the style of Link
